x-pack/plugins/alerting/server/lib/alerting_event_logger/alerting_event_logger.ts

```diff
@@ -393,6 +393,26 @@ export class AlertingEventLogger {
 
     this.eventLogger.logEvent(this.event);
   }
+
+  public reportGap({
```

Comment:
The format of the gap document looks good! I think it would also be useful to add the gap information to the `execute` document for the rule execution (maybe without the `status` info). If we do that, could we also assign a gap ID and have them in both docs?

Reply:
For what cases we want to have that?

Currently gap info stored in `gap` and `execution-metrics` documents. In theory we can have thuis information by execution uuid?

Reply:
We try to keep a holistic view of the rule run in the `execute` documents so we can avoid aggregating docs by the execution UUID when possible, so if the rule execution includes a gap, it would make sense to me to add it to the execute doc if it's straightforward. However, since we don't have any current plans to add this info to the stack rules UI, it's not strictly necessary at this point, so I'm fine with adding it in the future if we need it.